server/game/cards/attachments/01/bodyguard.js

```diff
@@ -1,17 +1,17 @@
 const DrawCard = require('../../../drawcard.js');
 
 class BodyGuard extends DrawCard {
-    setupCardAbilities() {
+    setupCardAbilities(ability) {
         this.interrupt({
             canCancel: true,
             when: {
                 onCharactersKilled: event => event.cards.includes(this.parent) && event.allowSave,
                 onCardsDiscarded: event => event.cards.includes(this.parent) && event.allowSave
             },
+            cost: ability.costs.sacrificeSelf(),
             handler: context => {
                 context.event.saveCard(this.parent);
```

Comment:
This will no longer work, because the `parent` is gone after the sacrifice cost executes.